src/javascript/crypto/e2e/openpgp/pgpmime/mimenode.js

```diff
@@ -130,10 +125,66 @@ e2e.openpgp.pgpmime.MimeNode.prototype.setContent = function(content) {
 
 
 /**
+ * Checks if the proposed boundary is contained within the message.
+ * @param {string} boundary The proposed boundary.
+ * @return {boolean} Returns true if the proposed boundary does not appear
+ *     within the message. Otherwise returns false.
+ * @private
+ */
+e2e.openpgp.pgpmime.MimeNode.prototype.isBoundaryValid_ = function(boundary) {
+  // Assign a temporary boundary that will be used in buildMessage().
+  this.boundary_ = 'temporary_boundary';
+  var tempMessage = this.mimeToString_();
+  return !goog.string.contains(tempMessage, boundary);
```

Comment:
For extra safety, check that the boundary is < 70 characters, as required by RFC 2046.